analyzer/src/main/kotlin/managers/utils/NuGetInspector.kt

```diff
@@ -0,0 +1,324 @@
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.analyzer.managers.utils
+
+import com.fasterxml.jackson.module.kotlin.readValue
+
+import java.io.File
+import java.util.SortedSet
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.decodeFromStream
+
+import org.ossreviewtoolkit.analyzer.PackageManager
+import org.ossreviewtoolkit.downloader.VersionControlSystem
+import org.ossreviewtoolkit.model.Hash
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.PackageReference
+import org.ossreviewtoolkit.model.Project
+import org.ossreviewtoolkit.model.ProjectAnalyzerResult
+import org.ossreviewtoolkit.model.RemoteArtifact
+import org.ossreviewtoolkit.model.Scope
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.VcsType
+import org.ossreviewtoolkit.model.yamlMapper
+import org.ossreviewtoolkit.utils.common.CommandLineTool
+import org.ossreviewtoolkit.utils.common.collectMessages
+import org.ossreviewtoolkit.utils.common.safeDeleteRecursively
+import org.ossreviewtoolkit.utils.ort.DeclaredLicenseProcessor
+import org.ossreviewtoolkit.utils.ort.createOrtTempFile
+import org.ossreviewtoolkit.utils.ort.showStackTrace
+
+private val json = Json { ignoreUnknownKeys = true }
+
+internal object NuGetInspector : CommandLineTool {
+    override fun command(workingDir: File?) = "nuget-inspector"
+
+    fun run(workingDir: File, definitionFile: File): Result {
+        val outputFile = createOrtTempFile(prefix = "nuget-inspector", suffix = ".json")
+
+        val commandLineOptions = buildList {
+            add("--project-file")
+            add(definitionFile.absolutePath)
+
+            add("--json")
+            add(outputFile.absolutePath)
+
+            add("--verbose")
+        }
+
+        return try {
+            run(workingDir, *commandLineOptions.toTypedArray())
+            outputFile.inputStream().use { json.decodeFromStream(it) }
+        } finally {
+            outputFile.delete()
+        }
+    }
+
+    private fun runNuGetInspector(definitionFile: File): Result {
+        val workingDir = definitionFile.parentFile
+
+        return runCatching {
+            try {
+                run(
+                    workingDir = workingDir,
+                    definitionFile = definitionFile,
+                )
+            } finally {
+                workingDir.resolve(".cache").safeDeleteRecursively(force = true)
+            }
+        }.onFailure { e ->
+            e.showStackTrace()
+
+            PackageManager.logger.error {
+                "Unable to determine dependencies for definition file '${definitionFile.absolutePath}': " +
+                        e.collectMessages()
+            }
+        }.getOrThrow()
```

Comment:
I find this construct with the nested `try` a bit strange.
How about:
```kt
return runCatching {
    run(
        workingDir = workingDir,
        definitionFile = definitionFile,
    )
}.also {
    workingDir.resolve(".cache").safeDeleteRecursively(force = true)
}.onFailure {
    it.showStackTrace()
    PackageManager.logger.error {
        "Unable to determine dependencies for definition file '${definitionFile.absolutePath}': " +
                it.collectMessages()
    }
}.getOrThrow()
```

Reply:
Interesting read in this context: https://stackoverflow.com/a/69756036/1127485

Reply:
so what do you guys prefer? We are not Kotlin experts?
I reckon that https://stackoverflow.com/a/69756036/1127485 states that try/finally is not the same as also?

Reply:
ah @TG1999 pushed the code update alright, so we are good now